hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/read/S3CachingInputStream.java

```diff
@@ -117,8 +120,8 @@ public void seek(long pos) throws IOException {
 
   @Override
   public void close() throws IOException {
-    super.close();
     this.blockManager.close();
+    super.close();
```

Comment:
why move it here? please add some comments if it is something others should be aware of.